src/Language/PureScript/Linter/Imports.hs

```diff
@@ -36,24 +36,34 @@ type UsedImports = M.Map ModuleName [Name]
 -- or references in an explicit import list.
 --
 findUnusedImports :: forall m. (Applicative m, MonadError MultipleErrors m, MonadWriter MultipleErrors m) => Module -> Env -> UsedImports -> m ()
-findUnusedImports (Module _ _ _ mdecls _) env usedImps = do
+findUnusedImports (Module _ _ _ mdecls mexports) env usedImps = do
   imps <- findImports mdecls
-  forM_ (M.toAscList imps) $ \(mni, decls) -> unless (mni `elem` autoIncludes) $
-    forM_ decls $ \(ss, declType, qualifierName) -> censor (onErrorMessages $ addModuleLocError ss) $
-      let usedNames = mapMaybe (matchName (typeForDCtor mni) qualifierName) $ sugarNames ++ M.findWithDefault [] mni usedImps in
-      case declType of
-        Implicit -> when (null usedNames) $ tell $ errorMessage $ UnusedImport mni
-        Explicit declrefs -> do
-          let idents = mapMaybe runDeclRef declrefs
-          let diff = idents \\ usedNames
-          unless (null diff) $ tell $ errorMessage $ UnusedExplicitImport mni diff
-        _ -> return ()
+  forM_ (M.toAscList imps) $ \(mni, decls) -> unless (mni `elem` alwaysUsedModules) $
+    forM_ decls $ \(ss, declType, qualifierName) ->
+      censor (onErrorMessages $ addModuleLocError ss) $ unless (qnameUsed qualifierName) $
+        let usedNames = mapMaybe (matchName (typeForDCtor mni) qualifierName) $ sugarNames mni ++ M.findWithDefault [] mni usedImps in
+        case declType of
+          Implicit -> when (null usedNames) $ tell $ errorMessage $ UnusedImport mni
+          Explicit declrefs -> do
+            let idents = mapMaybe runDeclRef declrefs
+            let diff = idents \\ usedNames
+            unless (null diff) $ tell $ errorMessage $ UnusedExplicitImport mni diff
+          _ -> return ()
   where
-  sugarNames :: [ Name ]
-  sugarNames = [ IdentName $ Qualified Nothing (Ident C.bind) ]
+  sugarNames :: ModuleName -> [ Name ]
+  sugarNames (ModuleName [ProperName n]) | n == C.prelude = [ IdentName $ Qualified Nothing (Ident C.bind) ]
```

Comment:
Sure, the difference between `negate` and `bind` is that this analysis happens after `negate` is desugared but before `bind` is desugared, by this point `negate` is correctly resolved to the appropriate fully qualified identifier (warning seems to work both ways as I just tested).

To  push the analysis later and do that properly would need to keep around the mapping of used names/their resolved fully qualified & original versions.